pkg/apis/core/v1alpha1/tiltrun_types.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2020 The Tilt Dev Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource"
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource/resourcestrategy"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TiltRun provides introspective data about the status of the Tilt process.
+// +k8s:openapi-gen=true
+type TiltRun struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   TiltRunSpec   `json:"spec,omitempty"`
+	Status TiltRunStatus `json:"status,omitempty"`
+}
+
+// TiltRunList is a list of TiltRun objects.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type TiltRunList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []TiltRun `json:"items"`
+}
+
+// TiltRunSpec defines the desired state of TiltRun
+type TiltRunSpec struct {
+	// TiltfilePath is the path to the Tiltfile for the run. It cannot be empty.
+	TiltfilePath string `json:"tiltfilePath"`
+	// ExitCondition defines the criteria for Tilt to exit.
+	ExitCondition ExitCondition `json:"exitCondition"`
```

Comment:
I'm wondering if going towards "Tilt as a daemon" I should change this verbiage to something like `DoneCondition` as it might not _necessarily_ trigger an "exit" in the traditional sense.

Reply:
ExitCondition seems ok to me. This is the Exit condition for _some_ process (for shell scripting), but we can be hand-wavey about which process that is for now